DEVELOPMENT.md

```diff
@@ -0,0 +1,106 @@
+2. Edit the appropriate `CHANGELOG.md` (in the appropriate package directory; e.g., `./federation-js/CHANGELOG.md`, `./gateway-js/CHANGELOG.md`), removing the `vNEXT` section entirely.  This section will remain on `main` and be resolved during the merge.
+3. Add a new section for `### vX.Y.Z` with a bullet indicating that something is coming soon:
+
+   ```
+   ### vX.Y.Z
+
+   - Nothing yet! Stay tuned.
+   ```
+
+4. Commit this change so GitHub will allow a PR to be opened against `main` with a notable change.  A suggested commit message is `Prepare CHANGELOG.md for upcoming vX.Y.Z release.`
+5. Push the branch to GitHub
+6. On GitHub, open a PR from the new release branch which targets `main`.
+   __For the title of the PR__, use "Release X.Y.Z" (See the "Note" above about how this might vary on this repository).  __For the body,__ use the contents of the template in the `.github/APOLLO_RELEASE_TEMPLATE.md` file in this repository.  Adjust the body as you see necessary.
+
+## Publishing a release
+
+### Step 1: Update the appropriate CHANGELOG.md files
+
+There is not a root `CHANGELOG.md` on this monorepo.  Instead, there are two main `CHANGELOG.md` files:
+
+- `./federation-js/CHANGELOG.md` for `@apollo/federation`
+- `./gateway-js/CHANGELOG.md` for `@apollo/gateway`.  __This is also the target CHANGELOG for changes to the `@apollo/query-planner` since that package is nottypically consumed/upgraded directly by our typical user.__
+
+Ensure that the appropriate CHANGELOG.md (using the guide above) within affected packages are up to date prior to bumping the version.  Additionally, it's best to go ahead and predict what the version is going to be published as in the next step and commit that in the CHANGELOG.  This allows the Git tags that will be created in Step 2 to include the changes.
+
+_Optional:_ To determine which packages are "changed" on the current branch (compared to the last published versions on the same branch; Lerna follows the `git tag`s on the branch to determine this, run the following from the project root:
+
+```
+npx lerna changed
+```
+
+This will list the packages that will have their versions bumped in the next step.
+
+### Step 2: Bump the version
+
+To bump the version, use the `release:version-bump` npm script.
+
+   __Option 1__: Bump all packages by the same version bump (e.g. `patch`, `minor`, `prerelease`, etc.).
+
+   > __Note__: Be sure to replace `<version-bump>` in the following command with the appropriate [version bump keyword](https://github.com/lerna/lerna/tree/f6e7a13e60/commands/version#semver-bump)
+
+   ```
+   npm run release:version-bump -- <version-bump>
+   ```
+
+   __Option 2__: Be interactively prompted for each new version.
+
+   This option works reasonably well in this federation repositiory since there are not all that many packages.  If no parameters are passed, a prompt will be displayed for each package asking for the new version.
```

Comment:
```suggestion
   This option works reasonably well in this federation repository since there are not all that many packages.  If no parameters are passed, a prompt will be displayed for each package asking for the new version.
```